site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,666 @@
+    }
+};
+
+const calculateCache = () => {
+    // Get cache recalculation request
+    cy.intercept('GET', buildUrl(['sample', 'bulk_late_days', 'calculate'])).as('calculateCache');
+
+    // Calculate all cache
+    cy.get('button').contains('Calculate Info').click();
+    cy.get('#rebuild-status-label').should('be.visible');
+
+    // Wait for query to finish
+    cy.wait('@calculateCache', {timeout: 300000});
+
+    // Wait for recalculation to finish
+    cy.get('#rebuild-status-label', {timeout: 15000}).should('not.be.visible');
+
+    for (const user_id of all_user_ids) {
+        cy.get(`[data-user="${user_id}"] > [data-before-content="Late Days Remaining"]`)
+            .then((cell) => expect(cell.text().trim()).not.to.equal(''));
+    }
+};
+
+// Test setup (sorted by submission due date)
+const test_info = [{
+    g_id: 'late_allowed_homework',
+    g_title: 'Late Allowed Homework',
+    info: '(1 day allowed)',
+    days_allowed: 1,
+},
+{
+    g_id: 'late_allowed_homework_2',
+    g_title: 'Late Allowed Homework 2',
+    info: '(2 days allowed)',
+    days_allowed: 2,
+}];
+
+// Object with user_ids that have late submissions for gradeables
+const all_late_users = {}; // {g_id: {user_id: #days_late}}
+// Grab all user IDs across all gradeables
+const all_user_ids = [];
+
+describe('Test cases involving late day cache updates', () => {
+    // Ignore uncaught js exceptions
+    Cypress.on('uncaught:exception', () => {
+        return false;
+    });
+
+    describe('Test accessing Bulk Late Days page as a student', () => {
+        it('should not allow access', () => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('student');
+            cy.get('.content').contains("You don't have access to this page");
+        });
+    });
+
+    describe('Test accessing Bulk Late Days as an instructor', () => {
+        it('should load properly', () => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('instructor');
+            cy.get('#late-day-table');
+        });
+    });
+
+    for (const test_case of test_info) {
+        // Grab testcase info
+        const { g_id, g_title, info, days_allowed } = test_case;
+        // Set up late submission information
+        all_late_users[g_id] = {};
+
+        describe(`Test students with late submissions ${info}`, () => {
+            it('should check students who have used too many late days for assignment', () => {
+                cy.visit(['sample', 'gradeable', g_id, 'grading', 'details']);
+                cy.login('instructor');
+                cy.get('[data-testid="view-sections"]').then(($button) => {
+                    if ($button.text().includes('View All')) {
+                        $button.click();
+                    }
+                });
+                let maximum_students = 0;
+                // Grab rows of students with late submissions (not NULL registration)
```

Comment:
I think it would be better to specify a few students who you know will have the specific late day requirements, rather than getting random students